website/docs/metadata.md

```diff
@@ -3,80 +3,173 @@ title: Metadata Table
+  optimized way, Hudi's metadata table improves query planning further by supporting multiple types of indices that aid 
+  in efficiently looking up data files based on relevant query predicates instead of reading the column stats from every 
+  individual data file and then pruning. 
+   
+## Supporting Multi-Modal Index in Hudi
+
+[Multi-modal indexing](https://www.onehouse.ai/blog/introducing-multi-modal-index-for-the-lakehouse-in-apache-hudi), 
+introduced in [0.11.0 Hudi release](https://hudi.apache.org/releases/release-0.11.0/#multi-modal-index), 
+is a re-imagination of what a general purpose indexing subsystem should look like for the lake. Multi-modal indexing is 
+implemented by enhancing Hudi's metadata table with the flexibility to extend to new index types as new partitions,
+along with an [asynchronous index](https://hudi.apache.org/docs/metadata_indexing/#setup-async-indexing) building 
+mechanism and is built on the following core principles:
+- **Scalable metadata**: The table metadata, i.e., the auxiliary data about the table, must be scalable to extremely 
+  large size, e.g., Terabytes (TB).  Different types of indices should be easily integrated to support various use cases 
+  without having to worry about managing the same. To realize this, all indices in Hudi's metadata table are stored as 
+  partitions in a single internal MOR table. The MOR table layout enables lightning-fast writes by avoiding synchronous 
+  merge of data with reduced write amplification. This is extremely important for large datasets as the size of updates to the 
+  metadata table can grow to be unmanageable otherwise. This helps Hudi to scale metadata to TBs of sizes. The 
+  foundational framework for multi-modal indexing is built to enable and disable new indices as needed. The 
+  [async indexing](https://www.onehouse.ai/blog/asynchronous-indexing-using-hudi) supports index building alongside 
+  regular writers without impacting the write latency.
+- **ACID transactional updates**: The index and table metadata must be always up-to-date and in sync with the data table. 
+  This is designed via multi-table transaction within Hudi and ensures atomicity of writes and resiliency to failures so that 
+  partial writes to either the data or metadata table are never exposed to other read or write transactions. The metadata 
+  table is built to be self-managed so users don’t need to spend operational cycles on any table services including 
+  compaction and cleaning    
+- **Fast lookup**: The needle-in-a-haystack type of lookups must be fast and efficient without having to scan the entire 
+  index, as index size can be TBs for large datasets. Since most access to the metadata table are point and range lookups,
+  the HFile format is chosen as the base file format for the internal metadata table. Since the metadata table stores 
+  the auxiliary data at the partition level (files index) or the file level (column_stats index), the lookup based on a 
```

Comment:
Should we also talk about record index here (i know it's already mentioned briefly in the next sextion below)? Record index is the most beneficial for needle in haystack problem if the predicate is on record key field. Also, with record index, Hudi now supports, indexing at partition, files and record level.

Reply:
@codope , these talk about how the multi modal indexing subsystem is built (using the HFile format for fast lookups on the internal metadata table.). This paragraph does not refer to workload categorization where we recommend different indices. Do you agree?

Reply:
sounds good.